beacon-chain/rpc/apimiddleware/custom_handlers.go

```diff
@@ -193,17 +201,53 @@ func handlePostSSZ(
 	return true
 }
 
-func sszRequested(req *http.Request) bool {
+func sszRequested(req *http.Request) (bool, error) {
 	accept, ok := req.Header["Accept"]
 	if !ok {
-		return false
+		return false, nil
+	}
+	if len(accept) == 0 {
+		return false, nil
 	}
-	for _, v := range accept {
-		if v == "application/octet-stream" {
-			return true
+	types := strings.Split(accept[0], ",")
+	// match a number with optional decimals
+	regex, err := regexp.Compile("q=\\d(\\.\\d)?")
+	if err != nil {
+		return false, err
+	}
+	currentType, currentPriority := "", 1.0
```

Comment:
Won't this cause the initial empty value to take precedence over a string where all values include a priority? I think this causes a bug; if you had an accept string like `application/octet-stream;q=0.95,application/json;q=0.9"`, the function would return false.

I think initializing `currentPriority` to `0` is going to simplify things. It would fix the bug just mentioned. You also won't need to check `if currentType == "" {` on line 227, because the priority comparison will implicitly handle that.